plugins/woocommerce/client/admin/client/settings-payments/onboarding/providers/woopayments/steps/payment-methods-selection/index.tsx

```diff
@@ -125,6 +129,35 @@ export default function PaymentMethodsSelection() {
 		}
 	};
 
+	// Check if overflow exists for Payment Methods list container.
+	const checkHasOverflow = () => {
+		const pmsContainer = scrollRef.current;
+
+		if ( pmsContainer ) {
+			// Compare scrollHeight and clientHeight to determine overflow.
+			setHasOverflow(
+				pmsContainer.scrollHeight > pmsContainer.clientHeight
+			);
+		}
+	};
+
+	// Check for overflow on initial render and on window resize.
```

Comment:
Can you share the reasoning behind checking the overflow here? I wonder if adding the border on top of the Continue button is easier, so we can skip this?

Reply:
That was the original implementation, Ahmed. However, following a Slack discussion (p1746612689367419-slack-C0800LPNCET), we decided to display the border conditionally only when the Payment Methods list container overflows. This was done to address inconsistent spacing when fewer Payment Methods are listed.